src/common/socket.cpp

```diff
@@ -373,22 +276,20 @@ bool _vsocket_final(void){
 }
 
 // hostname/ip conversion functions
-uint32 host2ip(const char* hostname)
-{
-	struct hostent* h = gethostbyname(hostname);
-	return (h != NULL) ? ntohl(*(uint32*)h->h_addr) : 0;
-}
-
-const char* ip2str(uint32 ip, char ip_str[16])
+std::string ip2str(uint32 ip)
```

Comment:
Death to `const char*` !  Long live `std::string` !

and you even remembered the nulls at the ends in the ShowWahetevrs :heart: